prow/cmd/checkconfig/main.go

```diff
@@ -143,28 +143,37 @@ func (o *options) Validate() error {
 	return nil
 }
 
-func gatherOptions() options {
```

Comment:
Can we get away with not changing this flow? Why is this better than the original impl?

Reply:
Because I need to pass the arguments that I want to check during a test. In the original implementation, `gatherOptions` reads from `os.Args`, so I don't have the chance to pass any argument. I do not think that the flow changed. It is just divided into two methods that I can unit test.